python/paddle/fluid/reader.py

```diff
@@ -364,6 +508,42 @@ def decorate_batch_generator(self, reader, places=None):
                 batched data.
             places (None|list(CUDAPlace)|list(CPUPlace)): place list. Must
                 be provided when PyReader is iterable.
+
+        Example:
+            .. code-block:: python
+
+                import paddle.fluid as fluid
+                import numpy as np
+
+                EPOCH_NUM = 3
+                ITER_NUM = 5
+
+
+                def random_image_generator(height, width):
+                    def generator():
+                        for i in range(ITER_NUM):
+                            yield np.random.uniform(low=0,
+                                                    high=255,
+                                                    size=[1, 1, height, width])
+
+                    return generator
+
+
+                image = fluid.layers.data(name='image', shape=[784, 784], dtype='float32')
+
+                reader = fluid.io.PyReader(feed_list=[image], capacity=4, iterable=True)
+
+                user_defined_generator = random_image_generator(784, 784)
+                reader.decorate_batch_generator(user_defined_generator,
```

Comment:
When using `decorate_batch_generator`, reader should yield a tuple/list of numpy array, whose first element is with shape of `[batch_size, 784, 784]`. Please fix it.

Reply:
Done